jre task fails because it doesn't set the appropriate dependencies on jar tasks

Given this project: https://github.com/hakanai/beryx-runtime-issue

If I try to run:

```shell
./gradlew :app:jre
```

The build fails with:

```
> Task :app:jre FAILED
4 actionable tasks: 4 executed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:jre'.
> File or directory not found: ...\beryx-runtime-issue\lib\build\libs\lib.jar
```

It seems like this task needs to add dependencies to all the jar tasks which it intends to look inside.

This is something we're seeing on our real gigantic project as well, but it seems all you need to reproduce it is a small project with at least two subprojects.

--- a/src/main/groovy/org/beryx/runtime/JreTask.groovy
+++ b/src/main/groovy/org/beryx/runtime/JreTask.groovy
@@ -63,6 +63,7 @@ class JreTask extends BaseTask {
 
     JreTask() {
         dependsOn('jar')
+        dependsOn(project.configurations.runtimeClasspath.allDependencies)
         description = 'Creates a custom java runtime image with jlink'
     }
 

